src/test/integration/git.test.ts

```diff
@@ -113,6 +113,11 @@ const makeFileChanges = async (repoDirectory: string) => {
     path.join(repoDirectory, "coverage", "foo", "bar"),
     "This file should be ignored",
   );
+  // Add a symlink in ignored directory
+  await fs.promises.symlink(
+    path.join(repoDirectory, "nested"),
+    path.join(repoDirectory, "node_modules", "nested")
+  );
```

Comment:
Wasn't sure how to test it, so I simply added a symlink in the `makeFileChanges` function in the test. Assumes `node_modules` is ignored in the test repo. I don't assert anything, but this should break the tests if the changes aren't present.

Reply:
I think i'd like to test this with a specific case that will fail when it doesn't handle symlinks properly. Will look into creating a test for it.